Source/Core/VideoCommon/NetPlayChatUI.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include <deque>
+#include <functional>
+#include <memory>
+#include <string>
+#include <utility>
+
+class NetPlayChatUI
+{
+public:
+  explicit NetPlayChatUI(std::function<void(const std::string&)> callback);
+  void Display();
+  void AppendChat(const std::string& message, const std::string& color);
+
+private:
+  char message_buf[160] = {};
+  bool scroll_to_bottom = false;
+  bool is_scrolled_to_bottom = true;
+
+  std::deque<std::pair<std::string, std::array<float, 3>>> messages;
```

Comment:
```suggestion
  std::deque<std::pair<std::string, std::array<float, 3>>> m_messages;
```

Also, why not just forward declare ImVec4 like  `struct ImVec4;` above the class?